src/bitcoin/bitcoinCoreRpc.ts

```diff
@@ -87,6 +88,12 @@ export class BitcoinCoreRpc implements BitcoinBlockchain {
     });
   }
 
+  public getBlockTime(): Promise<number> {
+    logger.info("Retrieving latest block");
+    const blockchaininfo = this.bitcoinClient.getblockchaininfo();
```

Comment:
:mega: This is an `async` call right?

Then this is not doing what you are think it is doing and will fail in production.
1. The function should be `async`
2. You will need an `await` at this line
3. The return value should be removed, otherwise you will have a `Promise<Promise<number>>` once you add the `async`.

Reply:
fixed, thanks 😃